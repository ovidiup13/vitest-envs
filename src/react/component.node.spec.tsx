import React from 'react';
import { describe, it, expect } from 'vitest';
import { renderToString } from 'react-dom/server';

import ExampleComponent from './component';

describe('SSR Component', () => {
  it('renders the browser message', () => {
    const result = renderToString(<ExampleComponent />);

    // We are expecting the useEffect NOT to execute and assert on the final state of the component.
    expect(result).toMatchSnapshot();
  });
});
