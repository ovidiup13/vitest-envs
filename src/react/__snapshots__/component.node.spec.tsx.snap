// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SSR Component > renders the browser message 1`] = `"<div>Hello, World!</div>"`;
